### ISSUE TITLE
Improve WhatsApp scrubbing for "<Media omitted>" logs

### ISSUE THREAD

**Issue by u-ways** [u-ways/xTractor#4]
Due to the nature of exported WhatsApp logs, chats that are exported "without media" will have the log message for any image obscured. This is an issue as some users will send an "x" with an image of exercise, and therefore this makes impossible to accuratly measure chats that were exported "without media".

Currently, a workaround has been implemented to count any WhatsApp's "\<Media omitted\>" logs found within a given day as a single "X" for each user. This means users will be rate limited to one visual "x" a day which is unfair as sending images is a great social motivator. **Note**: for media power users (users who send lots of "x" selfies) will have a scrubber that cancels above rate limiting, but this still makes the numbers inaccurate.

### Potential fixes/improvements:
- Use AI/NLP to have a "confidence score" of a given media omitted log that it is a potential "X" (leading or previous chats that are related to exercise can hint that it was an "x")
- The rate limit canceller / media power users scrubber should be rate limited (i.e. rewrite the scrubber to aggregate a "<Media omitted>" replacement log to a given amount instead of counting all "<Media omitted>" logs as exercises for media power users...)
  - Perhaps move this functionality to user report to not interfere with the quality of scrubbed data.
- Finally, if we keep noticing the data is too flawed when we derive points based on "<Media omitted>" lgos, Reject chat exports that have missing media messages. (i.e. user must export the chat with media but only collect the txt file export)
  - or, print out a warning indicating the report has "\<Media omitted\>" logs!

**Comment by smbkr** [comment]
Do you get a separate export with the media and any linked comments or only ‘<media omitted>’?

**Comment by u-ways** [comment]
@smbkr Only the media omitted message. You would only get a link to the image + the message in the log if you export with Media included.

So as of current when you export without Media:

```
[Date], [time] - [user]: \<Media omitted\>
```

But for the same message if you export with Media you get;


```
[Date], [time] - [user]: IMG-NNNN.jpg X - run!
```

Unfortunate limitation which made things hard for our group as we had a decent amount of text (19k) + we sent lots of images.

**Comment by u-ways** [comment]
Turns out they have further limitation on their export feature sadly: https://android.stackexchange.com/a/223971
It seems that the only surefire why to get the fully logs is to set a job to export in weekly bases due to cater for large media posted.

> Whatsapp export by mail has limitations. [WhatsApp FAQ](https://faq.whatsapp.com/en/android/23756533) says
If you choose to attach media, the most recent media sent will be added as attachments.
>
> When exporting with media, you can send up to 10,000 latest messages. Without media, you can send 40,000 messages. These constraints are due to maximum email sizes.
> 
> [This blog](https://www.zapptales.com/en/how-to-export-whatsapp-chat-android-iphone-ios/) adds more details
A maximum of the last 10.000 messages are sent. However, if you have some videos or images, the number of exported messages might be much smaller.
>
> WhatsApp will start creating a “package” with the latest Media files and text. It will put the last media files and messages first and then will stop when approximately 18 Mb of data have been packaged.
>
> If you have some videos in your chat, this could mean that you have two or three videos, some pictures and only one week of chat.
> 
> If you have basically many pictures, you will be able to export about 50 to 75 pictures and a few months of chat.
>
> In case that you do not have many media files, you will be able to export a maximum of 10.000 messages with this system
>
> I couldn't find any such limitations for export to Google drive in whatsapp FAQ or on my search on the net but going by this issue on [Google support](https://support.google.com/drive/thread/14988017?hl=en) and fair number of people who up voted this complaint, it seems to be common. As you can see, Google has washed their hands off and pointed to [WhatsApp](https://www.whatsapp.com/contact/). You may like to raise this issue with WhatsApp and post your findings here to cover the Google Drive portion

tl:dr; It's a WhatsApp limitation and there seems be to no workaround.
